lib/foreman/http_proxy/net_http_extension.rb

```diff
@@ -2,9 +2,8 @@ module Foreman
   module HttpProxy
     module NetHttpExtension
       def request(req, body = nil, &block)
-        host = URI.parse(@address).host
```

Comment:
Looking into Ruby code, it looks like host == `@address`.

https://github.com/ruby/ruby/blob/master/lib/net/http.rb#L671-L676

Reply:
Might this be an attempt to verify the address is a valid URI?

Reply:
Oh no, I failed to explain this miserably. Let me try again: at no point, `@address` contains a URI/URL. It contains either hostname or IP address. And the reason we see the error is that Ruby is a bit inconsistent here - it does parse invalid URI of `1.2.3.4` but it does not parse the same with IPv6 address:

```
>> URI.parse("1.2.4.5")
=> #<URI::Generic 1.2.4.5>
>> URI.parse("2a00:4bc0:1010:1102:0:b19:7175:babe")
Traceback (most recent call last):
        7: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'
        6: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `load'
        5: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        4: from (irb):2
        3: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/common.rb:234:in `parse'
        2: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:73:in `parse'
        1: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:67:in `split'
URI::InvalidURIError (bad URI(is not URI?): "2a00:4bc0:1010:1102:0:b19:7175:babe")
>> URI.parse("[2a00:4bc0:1010:1102:0:b19:7175:babe]")
Traceback (most recent call last):
        8: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'
        7: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `load'
        6: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        5: from (irb):2
        4: from (irb):3:in `rescue in irb_binding'
        3: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/common.rb:234:in `parse'
        2: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:73:in `parse'
        1: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/uri/rfc3986_parser.rb:67:in `split'
URI::InvalidURIError (bad URI(is not URI?): "[2a00:4bc0:1010:1102:0:b19:7175:babe]")
```